src/DistributedComputations/distributed_fft_based_poisson_solver.jl

```diff
@@ -196,12 +197,13 @@ function validate_global_grid(global_grid::RectilinearGrid)
 
     if (TY == Bounded && TZ == Periodic) || (TX == Bounded && TY == Periodic) || (TX == Bounded && TZ == Periodic)
         throw("NonhydrostaticModels on Distributed grids do not support topology ($TX, $TY, $TZ) at the moment.
-               TZ Periodic requires also TY and TX to be Periodic, while TY Periodic requires also TX to be Periodic.
-               Please rotate the domain to obtain the required topology")
+               A Periodic z-direction requires also the y- and and x-directions to be Periodic, while a Periodic y-direction requires also 
+               the x-direction to be Periodic.")
     end
     
     if !(global_grid isa YZRegularRG) && !(global_grid isa XYRegularRG) && !(global_grid isa XZRegularRG) 
-        throw(ArgumentError("Only stretching in one direction is supported with distributed grids at the moment."))
+        throw("The provided grid is stretched in directions $(stretched_dimensions(global_grid)). 
+               Distributed architectures support only RectilinearGrid that have variably-spaced cells in at most one direction.")
```

Comment:
still confused about this, because isn't this point specific to Poisson solvers for the nonhydrostatic model? There's no problem with grids at all. Why don't we want people to build grids?

Reply:
I myself was confused just reading the code because I don't understand why this code would end up in "validate global grid"... why can't we have global grids with these topologies?!??

Reply:
with regards to the stretching it is the same as for the serial algorithm. With regards to the topologies it has to do with the order of the transforms, you don't want to perform a r2r after a r2c transform. In the end, we can relax this requirement later because it requires a lot of additional code and multiple additional transposes so I think it is better just to not support it for the moment and add support incrementally.

The explanation should be quite clear now